### PR TITLE
bug 1155071 - configure log server

### PR DESCRIPTION
**Pull request by rhelmer**
r? @jdotpz @phrawzty

### CODE REVIEW

File: puppet/modules/socorro/manifests/role/common.pp [hunk 1]
```diff
@@ -8,4 +8,12 @@
         command => "/usr/bin/consul join ${consul_hostname}"
   }
 
+  file {
+    '/etc/rsyslog.d/conf.d/30-socorro.conf':
+      content => template('socorro/rsyslog.d/30-socorro.conf.erb'),
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0644',
+      notify  => Service[rsyslogd];
```
Comment (phrawzty) [inline puppet/modules/socorro/manifests/role/common.pp]: The service label should be `'single-quoted'`.

File: puppet/modules/socorro/templates/etc_rsyslog.d/30-socorro.conf.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#not set by default, needed to spool queues to disk
+$WorkDirectory /tmp
```
Comment (phrawzty) [inline puppet/modules/socorro/templates/etc_rsyslog.d/30-socorro.conf.erb]: Any time I see `/tmp/` I get nervous, hehe. What exactly is going to get put here? Pipes? Temp files?

File: terraform/README.md [hunk 3]
```diff
@@ -29,8 +29,10 @@ sensitive data.  The wrapper expects this bucket to have a read-writeable
 `tfstate/` directory at the root level.
 
 The secret bucket also contains hiera keys, in the `hiera/` directory at the
-root level. At minimum, you must specify the hostname of the ELB to the consul
-cluster, inside a file named `hiera/${environment}/consul_hostname`.
+root level. You must specify the hostname of the ELB to the consul
+cluster, inside a file named `hiera/${environment}/consul_hostname`, and the
+hostname to a syslog server inside a file named
```
Comment (phrawzty) [inline terraform/README.md]: These are two discrete configuration items: one deals with the Consul service, and the other with logging. Clumping them together here is expedient but potentially confusing. Suggest either splitting out into two distinct explanations, or (at least) modifying the line below to read something like
> Puppet will use this to ensure that all nodes automatically join the Consul and logging services, respectively.

File: puppet/modules/socorro/templates/etc_rsyslog.d/30-socorro.conf.erb (hunk 2 above)
Comment (rhelmer) [inline puppet/modules/socorro/templates/etc_rsyslog.d/30-socorro.conf.erb]: > Any time I see /tmp/ I get nervous, hehe. What exactly is going to get put here? Pipes? Temp files?

I'm not actually sure - I just used an existing file in `/etc/rsyslog.d/` as an example.

**Comment by rhelmer** [comment]
@jdotpz @phrawzty OK this has been tested and works in my env!

**Comment by jdotpz** [comment]
r+

File: puppet/modules/socorro/templates/etc_rsyslog.d/30-socorro.conf.erb (hunk 2 above)
Comment (jdotpz) [inline puppet/modules/socorro/templates/etc_rsyslog.d/30-socorro.conf.erb]: It's all good, this is where rsyslog queues logs to go to.  We won't want this to persist on a reboot, etc.

**Pull request merged**